app/src/pages/inside/profilePage/modals/deleteAccountFeedbackModal/deleteAccountFeedbackModal.jsx

```diff
@@ -0,0 +1,152 @@
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { defineMessages, useIntl } from 'react-intl';
+import classNames from 'classnames/bind';
+import { ModalLayout, withModal } from 'components/main/modal';
+import { COMMON_LOCALE_KEYS } from 'common/constants/localization';
+import { showModalAction } from 'controllers/modal';
+import { InputCheckbox } from 'components/inputs/inputCheckbox';
+import { Input } from 'components/inputs/input';
+import { FieldErrorHint } from 'components/fields/fieldErrorHint';
+import styles from './deleteAccountFeedbackModal.scss';
+
+const cx = classNames.bind(styles);
+const messages = defineMessages({
+  header: {
+    id: 'DeleteAccountFeedbackModal.header',
+    defaultMessage: 'Delete account',
+  },
+  description: {
+    id: 'DeleteAccountFeedbackModal.description',
+    defaultMessage: 'Why are you deleting your account?',
+  },
+  continue: {
+    id: 'DeleteAccountFeedbackModal.continue',
+    defaultMessage: 'Continue',
+  },
+  NO_NEEDED: {
+    id: 'DeleteAccountFeedbackModal.NO_NEEDED',
+    defaultMessage: 'The account is no longer needed',
+  },
+  DISSATISFIED: {
+    id: 'DeleteAccountFeedbackModal.DISSATISFIED',
+    defaultMessage: 'Dissatisfied with the service',
+  },
+  ALTERNATIVE: {
+    id: 'DeleteAccountFeedbackModal.ALTERNATIVE',
+    defaultMessage: 'Found a better alternative',
+  },
+  OTHER: {
+    id: 'DeleteAccountFeedbackModal.OTHER',
+    defaultMessage: 'Other',
+  },
+  inputError: {
+    id: 'DeleteAccountFeedbackModal.inputError',
+    defaultMessage: 'The field should have size not more than 128 symbols.',
+  },
+});
+
+const NO_NEEDED = 'NO_NEEDED';
+const DISSATISFIED = 'DISSATISFIED';
+const ALTERNATIVE = 'ALTERNATIVE';
+const OTHER = 'OTHER';
+
+const DeleteAccountFeedback = () => {
+  const { formatMessage } = useIntl();
+  const dispatch = useDispatch();
+  const [options, setOptions] = useState({
+    [NO_NEEDED]: false,
+    [DISSATISFIED]: false,
+    [ALTERNATIVE]: false,
+    [OTHER]: false,
+  });
+  const [otherValue, setOtherValue] = useState('');
+  const [error, setError] = useState('');
+
+  const disabledContinueButton = () => {
+    let isAnyOptionChecked = false;
+    Object.keys(options).forEach((key) => {
+      isAnyOptionChecked = isAnyOptionChecked || options[key];
+    });
+    return !isAnyOptionChecked || !!error;
+  };
+
+  const continueButton = {
+    text: formatMessage(messages.continue),
+    onClick: () => {
+      dispatch(
+        showModalAction({
+          id: 'deleteAccountModal',
+          data: {
+            options,
+            otherValue,
+          },
+        }),
+      );
+    },
+    disabled: disabledContinueButton(),
+  };
+  const cancelButton = {
+    text: formatMessage(COMMON_LOCALE_KEYS.CANCEL),
+  };
+
+  const onOtherValueChange = (e) => {
+    const currentValue = e.target.value;
+    setOtherValue(currentValue);
+    setOptions({ ...options, [OTHER]: !!currentValue });
```

Comment:
When new state is depends on prevState it is recommended to use updater function syntax